src/LibHdb++Cassandra.h

```diff
@@ -19,285 +24,325 @@
+        unsigned int ttl;
+    };
+
+    CassCluster *mp_cluster;
+    CassSession *mp_session;
+    string m_keyspace_name;
+    map<string, struct ConfParams> attr_ID_map;
+    bool logging_enabled;
+    CassLogLevel cassandra_logging_level;
 
-	enum FindAttrResult { AttrNotFound, FoundAttrWithDifferentType, FoundAttrWithSameType };
-	enum extract_t { EXTRACT_READ, EXTRACT_SET };
+public:
+    /**
+     * @brief HdbPPCassandra destructor
+     *
+     * The destructor will attempt to disconnect an open cassandra session
+     */
+    ~HdbPPCassandra();
 
-	struct ConfParams
-	{
-		CassUuid id;
-		unsigned int ttl;
-	};
+    /**
+     * @brief HdbPPCassandra constructor
+     *
+     * The configuration parameters must contain the following strings:
+     *
+     * - Mandatory:
+     *     - contact_points: Cassandra cluster contract host, eg cassandra_db
+     *     - keyspace: Keyspace to use within the cluster, eg, hdb_test
+     * - Optional:
+     *      - user: Cluster log in user name
+     *      - password: Password for above user name
+     *      - local_dc: Datacenter name
+     * - Debug:
+     *     - logging_enabled: Either true to enable command line debug, or false to disable
+     *     - cassandra_driver_log_level  Cassandra logging level, see CassLogLevel in Datastax
+     * documentation. This
+     *       must be one of the follow values:
+     *          - TRACE: Equivalant CASS_LOG_TRACE
+     *          - DEBUG: Equivalant CASS_LOG_DEBUG
+     *          - INFO: Equivalant CASS_LOG_INFO
+     *          - WARN: Equivalant CASS_LOG_WARN
+     *          - ERROR: Equivalant CASS_LOG_ERROR
+     *          - CRITICAL: Equivalant CASS_LOG_CRITICAL
+     *          - DISABLED: Equivalant CASS_LOG_DISABLED
+     *
+     * @param configuration A list of configuration parameters to start the driver with.
+     */
+    HdbPPCassandra(vector<string> configuration);
 
-	CassCluster* mp_cluster;
-	CassSession* mp_session;
-	string m_keyspace_name;
-	map<string,struct ConfParams> attr_ID_map;
-	bool logging_enabled;
-	CassLogLevel cassandra_logging_level;
+    /**
+     * @brief Insert an attribute insert event into the database
+     *
+     * Inserts an attribute insert event for the EventData into the database. Neither empty
+     * or invalid attributes will be archived, and if the attribute does not exist an
+     * exception will be raised.
+     *
+     * @param data Tango event data about the attibute.
+     * @param ev_data_type HDB event data for the attibute.
+     * @throw Tango::DevFailed
+     */
+    virtual void insert_Attr(Tango::EventData *data, HdbEventDataType ev_data_type);
 
-public:
-	~HdbPPCassandra();
-	HdbPPCassandra(vector<string> configuration);
+    /**
+     * @brief Inserts an attibute parameter event into the database
+     *
+     * Inserts the parameter data (attribute event data) about an existing attribute
+     * into the database. The attribute must exist, otherwise an exception is thrown.
```

Comment:
To be fair, I have found the API functon naming a little confusing and at times disconnected from what the functions actually do. Its been really helpful to have my comments reviewed to improve my understanding here.